### ISSUE TITLE
Master List: Cliffnote Topic Requests

### ISSUE THREAD

**Issue by nesquena** [codepath/android_guides#2]
This is the **master list** of things that need to be added / improved on the cliffnotes. If you see an **important topic missing** here, **please reply below**!

Incomplete stubs that **need attention** to be completed (please help out if you can):
- [Advanced Architecture Guides](http://guides.codepath.com/android/Architecture-of-Android-Apps)
- [Profiling Performance or Memory](https://github.com/codepath/android_guides/wiki/Debugging-and-Profiling-Apps)
- [Real-time Chat with Push](https://github.com/codepath/android_guides/wiki/Real-time-Messaging)
- [Drawing with SurfaceView](https://github.com/codepath/android_guides/wiki/Extending-SurfaceView)
- [Bluetooth API](https://github.com/codepath/android_guides/wiki/Connectivity-using-the-Bluetooth-API)

A few topics we could use help developing cliffnotes for: 
- [Wearables Guides](https://developer.android.com/training/building-wearables.html)
- [WiFi](http://developer.android.com/reference/android/net/wifi/WifiManager.html), [NFC](https://github.com/codepath/android_guides/issues/23) Connectivity APIs
- [Creating an App Widget](https://github.com/codepath/android_guides/issues/40) (#40)
- [Google Play In-app Billing](http://developer.android.com/google/play/billing/index.html) (#27)
- [Custom Camera with Camera 2 API](https://github.com/codepath/android_guides/issues/86)
- [AccountManager](https://github.com/codepath/android_guides/issues/49) (#49)
- [AsyncTaskLoader](http://developer.android.com/reference/android/content/AsyncTaskLoader.html)
- [SyncAdapter](http://developer.android.com/training/sync-adapters/creating-sync-adapter.html)
- [Lollipop Job Scheduler](https://github.com/codepath/android_guides/issues/74) (#74)
- [Geofencing API](https://github.com/codepath/android_guides/issues/192) (#192)

Production Topics:
- [Building Responsive Android Apps](https://github.com/codepath/android_guides/issues/55)
- [Checklist on preparing for public release](https://github.com/codepath/android_guides/issues/69)
- [I18n and Localization guide](https://github.com/codepath/android_guides/issues/77)
- [Guide on beta testing in lead up to launch](https://github.com/codepath/android_guides/issues/121)

Workflow Tools:
- Crashlytics
- Stripe (#73)
- OTA
- Facebook SDK
- Updated Jenkins
- Advanced Gradle (Flavors, Tasks, Deps)

More advanced topics that could use cliffnotes by experts:
- Creating a Custom Keyboard (#137)
- Security for Android 
- Native Development Kit (NDK)
- Binder Inter-Process Communications ([See Here](https://guides.codepath.com/android/Managing-Threads-and-Custom-Services#bound-services))

Any help starting these cliffnotes would be appreciated! Feel free to add other requests as an issue or to this thread! If you are interested in contributing, please [check out the contributing guidelines](https://github.com/codepath/android_guides/wiki/Contributing-Guidelines) to understand the philosophy of the guides.

**Comment by fanky10** [comment]
I think Creating and Using Providers is pretty much updated:
https://github.com/codepath/android_guides/wiki/Creating-Content-Providers

Would you like to remove it from the list?

**Comment by nesquena** [comment]
@fanky10 Yes, you are right. Thanks to @AdamMc331 and @princessruthie, the content providers guide can now be removed. Thanks you too!

**Comment by Amejia481** [comment]
It will be great to have a guide about [Exoplayer](https://github.com/google/ExoPlayer).

**Comment by Alifyz** [comment]
Hi Guys, 
What about Firebase? 
Do you guys think that would be interesting to have one article about it?

**Comment by fabricethilaw** [comment]
Yes, it would be useful for me and super interesting to do it ( for Android
app as far I am concerned :) ). Because it seems incredibly simple when you
run the too easy samples provided on Github. But as you go deeper and try
to incorporate Firebase into a complex app logic, then you get trouble with
data structure, read and write codes.

*Thilaw Fabrice KIKI*

Software developer for mobile devices


My achievements on Google Play Store <https://goo.gl/8CWQoL>


Twitter : @ThilawFab <https://twitter.com/ThilawFab>

**Comment by josephsarz** [comment]
> It will be great to have a guide about [Exoplayer](https://github.com/google/ExoPlayer).

please let there be an [Exoplayer](https://github.com/google/ExoPlayer) guide

**Comment by root-ansh** [comment]
Hey @codepath I have been messing around with exoplayer for a while now. I am not an expert but can Contribute whatever i know, in the spirit of #hacktoberfest. Should I add that that to [Video-Playback-and-Recording](https://guides.codepath.com/android/Video-Playback-and-Recording) guide?

**Comment by nesquena** [comment]
Yeah take a look at https://guides.codepath.com/android/Playing-Media-with-ExoPlayer-or-TubiPlayer and see what you think too

**Comment by root-ansh** [comment]
oops apologies i didn't find that page so added it in the video player and recording guide, hope that's not a problem

**Comment by nesquena** [comment]
@root-ansh No problem at all, thanks for contributing!  I moved your guide to a separate page here: https://github.com/codepath/android_guides/wiki/Quickstart-with-Exoplayer and provide links to both.

**Comment by root-ansh** [comment]
glad i could help

**Comment by Tonymunyao12** [comment]
Hello @codepath-adminstaff,

I appreciate the effort you guys have put in helping other Android Developers in accessing quality content.

I would really like to be part of the contributing team. I will check on which topics I can add to the main android-guides.

Thank you for your consideration.
